packages/evm/contracts/budgets/ManagedBudgetWithFees.sol

```diff
@@ -213,11 +205,21 @@ contract ManagedBudgetWithFees is AManagedBudgetWithFees, ManagedBudget {
     ) internal {
         address incentive = address(boost.incentives[incentiveId]);
         uint256 maxAmount = incentiveFees[incentive];
-        uint256 amount = maxAmount * claim / limit;
-        delete incentiveFees[incentive];
+        uint256 alreadyPaid = incentiveFeesDisbursed[incentive];
+        uint256 amount = maxAmount * claim / limit - alreadyPaid;
+        incentiveFeesDisbursed[incentive] = alreadyPaid + amount;
         address asset = AIncentive(incentive).asset();
         address manager = boost.owner;
-        reservedFunds[asset] -= maxAmount;
+        reservedFunds[asset] -= amount;
+
+        if (amount == 0) {
+            revert BoostError.ZeroBalancePayout();
+        }
+
+        if (alreadyPaid + amount == maxAmount) {
```

Comment:
Does it matter if we reset these or not?

Reply:
It's not technically, but it's good citizenship to delete the state.